### ISSUE TITLE
Pop!_OS 20.04: Display Recognized but not usable

### ISSUE THREAD

**Issue by brandoningli** [DisplayLink/evdi#279]
* Are you using the latest driver? Yes, 5.4.0-55.153
    
  * Are you using the latest EVDI version? Yes, 1.9.1
    
  * If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website? Yes
    
  * Is this issue related to evdi/kernel? Assuming so
    

  * Linux distribution and its version: Pop!_OS 20.04 LTS
  * Linux kernel version: 5.11.0-7612-generic
  * Xorg version (if used): `xserver-xorg-core` version 1.20.9-2ubuntu1.2~20.04.2
  * Desktop environment in use: GNOME 3.36.8

Using a Dell USB3 docking station. All other functions (USB, Ethernet) work fine. Display will show up in the GNOME settings with the correct metadata, and I can ask GNOME to 
join displays, but it immediately sets back to single display on my laptop's built-in monitor.

**Comment by brandoningli** [comment]
When attempting with `arandr`, this error appears:
```
xrandr: Configure crtc 6 failed
```

**Comment by displaylink-emajewsk** [comment]
Hi Brandon. 😇 

I didn't get much time to investigate this issue yet. However, everything appears to be fine on Wayland, if you don't mind switching to it as a workaround. Here's a helpful guide for enabling Wayland on Pop!_OS: [https://www.reddit.com/r/pop_os/comments/75wn18/how_do_i_start_wayland_session/doag5r8/](https://www.reddit.com/r/pop_os/comments/75wn18/how_do_i_start_wayland_session/doag5r8/?utm_source=reddit&utm_medium=web2x&context=3)

**Comment by brandoningli** [comment]
I have an NVidia chip that I use often off-dock, so since I don't want to be hopping between it and xorg, using Wayland as a workaround isn't really an option for me. I have been connecting to my monitor directly with my laptop's built-in HDMI port in the meantime.

**Comment by displaylink-emajewsk** [comment]
Ah, that's unfortunate. 😣 I'll be taking a closer look at it in the near future.

**Comment by neineisansan** [comment]
Just made an account to drop in and say I'm basically in exactly the same boat on system76 laptop very similar setup. X11 w/ Nvidia. Xrandr was recognizing the display when connected but it had a black screen and would show up in the display settings but if I tried to extend my display to it it would just pop back over to my default display. I've since uninstalled displaylink/evdi but oh well that's what I get for updating my kernel lol.

**Comment by brentpinkston** [comment]
I am also in the same boat, brand new laptop and after 2 days of using it the 5.11 Kernel comes out and my 2nd Monitor quits working.  I ended up going back to the 5.8 Kernel, and it is fine, I just have to be careful what software I update.

**Comment by bolek2000** [comment]
Hi I have the same problem, I uninstalled DL 5.4, switched back to kernel Version 5.8 and reinstalled DL 5.3, but now my old configuration does'nt work either. Not sure why. In the release notes it said something about firmware upgrades for the chips, does that play a role ? I use an i-tec docking station.

```
Linux cs-nb 5.8.0-7642-generic #47~1614007149~20.04~82fb226-Ubuntu SMP Tue Feb 23 02:56:27 UTC  x86_64 x86_64 x86_64 GNU/Linux

DKMS status:
evdi, 1.7.0, 5.8.0-7642-generic, x86_64: installed
nvidia, 460.67, 5.11.0-7612-generic, x86_64: installed
nvidia, 460.67, 5.8.0-7642-generic, x86_64: installed
system76, 1.0.12~1616424713~20.04~c42d494, 5.11.0-7612-generic, x86_64: installed
system76, 1.0.12~1616424713~20.04~c42d494, 5.8.0-7642-generic, x86_64: installed
system76_acpi, 1.0.2~1600812457~20.04~0bc966c, 5.11.0-7612-generic, x86_64: installed
system76_acpi, 1.0.2~1600812457~20.04~0bc966c, 5.4.0-7642-generic, x86_64: installed
system76_acpi, 1.0.2~1600812457~20.04~0bc966c, 5.8.0-7630-generic, x86_64: installed
system76_acpi, 1.0.2~1600812457~20.04~0bc966c, 5.8.0-7642-generic, x86_64: installed
system76-io, 1.0.1~1616424744~20.04~57c15ff, 5.11.0-7612-generic, x86_64: installed
system76-io, 1.0.1~1616424744~20.04~57c15ff, 5.8.0-7642-generic, x86_64: installed

dmsg:
[  177.084908] evdi: [D] evdi_painter_connect:738 (dev=0) Process is trying to connect
[  177.084910] evdi: [I] Task 4279 (DesktopManagerE) of process 4272 (DisplayLinkMana)
[  177.084912] evdi: [D] evdi_painter_connect:788 (dev=1) Connected with 0000000053f0fac0
[  177.084914] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  177.092197] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.092210] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.092249] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.092259] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.092292] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.092302] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.092333] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  177.092334] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  177.092585] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  177.092586] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  177.252723] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.252733] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.252740] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  177.252751] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  177.252752] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  178.009090] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  178.009106] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  178.009118] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  178.009137] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  178.009138] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  178.036656] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  178.036667] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  178.036674] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  178.036686] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  178.036687] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  180.158615] cdc_ncm 2-2.2:1.5 enx803f5d103f61: 1000 mbit/s downlink 1000 mbit/s uplink
[  180.190617] cdc_ncm 2-2.2:1.5 enx803f5d103f61: network connection: connected
[  180.190643] IPv6: ADDRCONF(NETDEV_CHANGE): enx803f5d103f61: link becomes ready
[  189.712737] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  189.712760] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  189.712777] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  189.712797] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  189.712798] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  190.194055] usb 1-2.1.2: reset high-speed USB device number 10 using xhci_hcd
[  206.764043] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.764055] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.764062] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.764074] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  206.764075] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  206.779259] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.779271] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.779278] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.779290] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  206.779291] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  206.904585] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.904596] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.904603] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  206.904615] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  206.904616] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  210.721353] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.721365] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.721373] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.721386] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  210.721387] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  210.801152] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.801162] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.801168] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.801180] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  210.801181] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  210.878600] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.878610] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.878617] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.878629] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  210.878630] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  210.952914] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.952925] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.952931] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  210.952944] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  210.952945] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  211.029786] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  211.029797] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  211.029804] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  211.029817] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  211.029818] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid
[  211.069965] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  211.069976] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  211.069983] evdi: [D] evdi_detect:94 (dev=0) poll connector state: disconnected
[  211.069995] evdi: [D] evdi_detect:90 (dev=1) poll connector state: connected
[  211.069996] evdi: [D] evdi_painter_get_edid_copy:228 (dev=1) EDID valid

```

**Comment by timnolte** [comment]
Yep, as others I'm on a System76 machine and now video on my dock is effectively dead. ☹️ Hopefully this can get resolved soon.

**Comment by elguero** [comment]
Perhaps this is good news.  It looks like NVIDIA support using XWayland is coming.  This was merged last week: https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/587

**Comment by echosa** [comment]
I have this issue on a System76 Darter Pro (darp5 model) which only has Intel graphics (no Nvidia or AMD) running Pop!_OS 20.10. I'm using DisplayLink for my SideTrak portable monitor. https://standsteady.com/products/sidetrak 

Similar to @bolek2000, I tried going to back to kernel 5.8 and driver version 5.3, but that no longer works either. 

Unfortunately, the laptop broke, and I'm getting it repaired under warranty, so I won't have access to it to do any testing or provide any more information for now. However, I wanted to make sure my case was documented as well.

**Comment by echosa** [comment]
@displaylink-emajewsk Just curious if there's any news on this. Thanks!

**Comment by binkocd** [comment]
Weighing in as well. System76 Darter Pro (darp5, Intel UHD 620), unable to use dock for a few weeks now. Tried going back to previous kernels, drivers, etc, no go. Interested in helping any way I can. Please let me know if there are any commands, outputs, etc that might help, as I would love to use my dock again.

**Comment by cement-head** [comment]
Running Ubuntu 20.04 on a System76 Galago Pro (glap3-c) with Intel UHD WL2 Graphics.  Using the System76 Drivers and second display is unusable with the dock, have to connect discretely with an HDMI cable.  I believe the issue is an incorrect (unpatched) EVDI 1.9.1 that is being bundled with DLM 5.4. https://github.com/DisplayLink/evdi/issues/275

**Comment by elguero** [comment]
The current release of EVDI, 1.9.1, is what is in the DLM 5.4 release.

**Comment by omad** [comment]
I'm running Pop_OS 20.10 and my displaylink dock connected monitors stopped working too, but I was able to get it working again by following [the instructions in this reddit post](https://www.reddit.com/r/pop_os/comments/mrl4j7/has_the_bluetooth_and_other_bugs_with_the_511/gwboavj/?utm_source=share&utm_medium=web2x&context=3) :

1. Roll-back mesa to 20.2.6:
```
sudo apt install mesa-va-drivers=20.2.6-0ubuntu0.20.10.1 mesa-vdpau-drivers=20.2.6-0ubuntu0.20.10.1 mesa-vulkan-drivers=20.2.6-0ubuntu0.20.10.1 libegl-mesa0=20.2.6-0ubuntu0.20.10.1 libegl1-mesa=20.2.6-0ubuntu0.20.10.1 libegl1-mesa-dev=20.2.6-0ubuntu0.20.10.1 libgl1-mesa-dri=20.2.6-0ubuntu0.20.10.1 libgl1-mesa-glx=20.2.6-0ubuntu0.20.10.1 libglapi-mesa=20.2.6-0ubuntu0.20.10.1 libglx-mesa0=20.2.6-0ubuntu0.20.10.1 libosmesa6=20.2.6-0ubuntu0.20.10.1 libgl1-mesa-dev=20.2.6-0ubuntu0.20.10.1 libgbm1=20.2.6-0ubuntu0.20.10.1
```
2. Install kernel 5.8.18 (I used the [ubuntu mainline tool](https://github.com/bkw777/mainline#downloads--source-code))

3. Switch your kernel:

`sudo kernelstub -v -k vmlinuz-5.8.18-050818-generic -i initrd.img-5.8.18-050818-generic`

4. Reboot.. I also had to re-enable one of my monitors using the Display Settings widget.

**Comment by timnolte** [comment]
@omad my only concern with doing this is that I am actually running on a System76 machine and don't 2amt to lose any hardware specific fixes/optimizations with the newer kernel.

**Comment by echosa** [comment]
@displaylink-emajewsk Has there been any progress figuring out the issue with Xorg?

**Comment by cement-head** [comment]
@omad Are you saying that MESA 21.0.0 is broken as well?

```
$ glxinfo | grep "OpenGL version"
OpenGL version string: 4.6 (Compatibility Profile) Mesa 21.0.0
```
Or will MESA 21.0.0 not work with Kernel 5.8.x ?

**Comment by cement-head** [comment]
I'm wondering, is this a bug in MESA 21, or a bug in DLM/EVDI? Has anyone reported this upstream to MESA?

**Comment by cement-head** [comment]
Has anyone used this script with any success?

displaylink-debian: https://github.com/AdnanHodzic/displaylink-debian

**Comment by echosa** [comment]
@cement-head I just tried that. It makes it to the point where it's configuing Xorg, then freezes the GUI. I can move the mouse and switch to other TTY's (ctrl-alt-F3), but I can't interact with anything in GNOME. I rebooted from the F3 TTY.
![2021-05-26 10-04-08](https://user-images.githubusercontent.com/133786/119685196-c476c300-be0a-11eb-927e-0013c3430cac.jpeg)

Possibly related to this issue since the installer also asked to install linux-source for me: https://github.com/AdnanHodzic/displaylink-debian/issues/581

**Comment by echosa** [comment]
Could we please get some of update on this? If there's anything we can do to help, I'm sure at least some of us would be happy to do so. @displaylink-emajewsk @displaylink-dkurek @displaylink-mlukaszek @DisplayLink-Admin

**Comment by cement-head** [comment]
@echosa That's exactly what happened to me - EXACTLY!

**Comment by cement-head** [comment]
Bought this: https://www.amazon.com/Docking-MCY-Displayport-Ethernet-Compatible/dp/B08L4RK1M7

Works, with no drivers - Ubuntu 20.04, Kernel 5.11, System76 PPA

**Comment by timnolte** [comment]
> Bought this: https://www.amazon.com/Docking-MCY-Displayport-Ethernet-Compatible/dp/B08L4RK1M7
> 
> Works, with no drivers - Ubuntu 20.04, Kernel 5.11, System76 PPA

Yes, but when you've already dropped money on a docking station that was working the solution shouldn't be to have to buy something else.

**Comment by brandoningli** [comment]
> Bought this: https://www.amazon.com/Docking-MCY-Displayport-Ethernet-Compatible/dp/B08L4RK1M7
> 
> Works, with no drivers - Ubuntu 20.04, Kernel 5.11, System76 PPA

This also doesn't help those of us who are using a USB-A DisplayLink docking station since we don't have a type-c port.

**Comment by cement-head** [comment]
@timnolte @brandoningli I agree with both of you guys - I'm frustrated with the lack of willingness on Synaptics part (DL) to concede that they need to be _ahead_ of kernel releases, not six months behind them.  Guess my point is that there is hardware that is fully compatible with the LINUX kernel that doesn't need a driver.  Synaptics/DL should open source their driver and push it into the LINUX kernel.

**Comment by mosqueteiro** [comment]
Have this issue as well —monitor is detected but not able to display anything on it. Dock and monitor works just fine on the windows boot and with my MacBook Pro. I have tried official installer and the displaylink-debian: https://github.com/AdnanHodzic/displaylink-debian

Not sure what else to try...

```
--------------- Linux system info ----------------

Distro: Pop
Release: groovy
Kernel: 5.11.0-7614-generic

---------------- DisplayLink info ----------------

Driver version: 1.7.0
1.9.1
DisplayLink service status: up and running
EVDI service version: 1.9.1

------------------ Graphics card -----------------

Vendor: i915
Subsystem: 630
VGA: Intel Corporation HD Graphics 630 (rev 04)
VGA (3D): NVIDIA Corporation GP107M [GeForce GTX 1050 Mobile] (rev a1)
X11 version: 1.20.9-2ubuntu1.3
X11 configs: /etc/X11/xorg.conf.d/20-displaylink.conf

-------------- DisplayLink xorg.conf -------------

File: /etc/X11/xorg.conf.d/20-displaylink.conf
Contents:
 Section "Device"
	Identifier "DisplayLink"
	Driver     "modesetting"
	Option     "PageFlip" "false"
EndSection

-------------------- Monitors --------------------

Providers: number : 6
Provider 0: id: 0x45 cap: 0x9, Source Output, Sink Offload crtcs: 3 outputs: 3 associated providers: 4 name:modesetting
Provider 1: id: 0x16c cap: 0x6, Sink Output, Source Offload crtcs: 1 outputs: 1 associated providers: 1 name:modesetting
Provider 2: id: 0x14a cap: 0x6, Sink Output, Source Offload crtcs: 1 outputs: 1 associated providers: 1 name:modesetting
Provider 3: id: 0x128 cap: 0x6, Sink Output, Source Offload crtcs: 1 outputs: 1 associated providers: 1 name:modesetting
Provider 4: id: 0xf3 cap: 0x6, Sink Output, Source Offload crtcs: 1 outputs: 1 associated providers: 1 name:modesetting
Provider 5: id: 0xd1 cap: 0x0 crtcs: 0 outputs: 0 associated providers: 0 name:modesetting
```

**Comment by echosa** [comment]
The [previous instructions|https://github.com/DisplayLink/evdi/issues/279#issuecomment-843282382] stopped working for me, so now I'm back to using the latest kernel and packages and hoping this gets fixed soon. I might try Wayland again, since the monitor works there. Unfortunately, using Wayland breaks some of my existing tools and programs I use, like AutoKey. :-(

**Comment by lspintzyk** [comment]
Guys,
As one of you has observed, mesa is required to be roll back.
DisplayLink displays are black with Mesa 21.0
Previous release of Mesa 20.3.X should be fine.
You can check what Mesa you have with:
glxinfo | grep "OpenGL version"

This is because of GL renderer in Mesa 21 has changed from llvmpipe to zink, and this made XServer go completly new path which is at the moment not working.

**Comment by cement-head** [comment]
If you're MESA (**OpenGL renderer**) is running on llvmpipe, it's running on software rendering as opposed to hardware rendering, and it's basically not installed properly.  You should be able to get it MESA/OpenGL to run directly on your GPU (dGPU, iGPU or eGPU).

```
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) UHD Graphics 620 (WHL GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 21.0.0
```

**Comment by Ohtar10** [comment]
I'm not sure how much my experience will contribute but here I will share what happened to me:

I use a System 76 AdderWS with a GeForce RTX 2070 (driver version 460.73.01), Pop OS 20.04 with kernel 5.11.0-7614-generic. I tried to use a Dell Universal Dock D6000 (https://www.dell.com/en-us/shop/accessories/apd/452-BCYT?c=us&l=en&s=gen&sku=452-BCYT). I installed the latest DisplayLink drivers at the moment (5.4.0-55.153) and rebooted. This totally broke my laptop, after reboot, it got stuck loading modules and it never gave me access, not even while unplugged from the docking station. The only way to recover my laptop was to forcefully access the terminal and uninstall the DisplayLink drivers.

So in conclusion from my experience:

1. Without drivers installed, the docking station simply does not work.
2. With drivers installed the laptop itself breaks and turns unusable unless I uninstall the DL drivers.
3. The docking station is completely useless in my case.

IMHO, If the hardware is not compatible with my setup and I'm unable to use the docking station, well that's a shame, sometimes happens and I'm unlucky. BUT the fact that the drivers completely break the laptop is completely crazy and IT SHOULD NOT HAPPEN.

**Comment by lspintzyk** [comment]
Related Xorg-xserver issue:
https://gitlab.freedesktop.org/xorg/xserver/-/issues/1183
and merge request with a fix
https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/681

**Comment by echosa** [comment]
I wonder how long it will take that change to make its way into the apt repos for Pop!_OS once its merged. Hopefully, it's not a "wait for 21.10 or later" kinda thing.

**Comment by lspintzyk** [comment]
> 
> 
> I wonder how long it will take that change to make its way into the apt repos for Pop!_OS once its merged. Hopefully, it's not a "wait for 21.10 or later" kinda thing.

Bug with fix porting request is here:
https://bugs.launchpad.net/ubuntu/+source/xorg-server/+bug/1931547

If you want to have it faster then lobby it:)

**Comment by cement-head** [comment]
Okay - everyone on this thread should cross post to the Ubuntu Bug report and let them know that this affects a lot of people!

**Comment by cement-head** [comment]
This looks as if it's fixed, but the XOrg Server needs to be built for Ubuntu 20.04: https://bugs.launchpad.net/ubuntu/+source/xorg-server/+bug/1931547

Can someone do this, or provide instructions?

**Comment by consentfactory** [comment]
FWIW, I'm on PopOS 20.04 and a workaround for this is to switch to wayland versus xorg (this worked for me).

The gist: comment out the following in `/etc/gdm3/custom.conf`:

`WaylandEnable=false`

Reboot, then at login _before you login_ click on the gear icon on the bottom right and select 'PopOS on Wayland'. 

The caveat here is that wayland is unstable on some GPUs. However, I'm on all Intel machine on a Dell laptop and its buttery smooth. YMMV.

**Comment by echosa** [comment]
Seems to be working in X11 after the 21.04 update! After I upgraded, my second display wasn't working at all. I uninstalled the driver, rebooted, installed the driver again, rebooted, and now it's working in X11! Not sure why, but I'm not going to question it. :-D

**Comment by cement-head** [comment]
@echosa It's likely the point release XOrg XServer 1.20.11 (Hirsute) vs 1.20.9 (Focal), that has the fix from the master.  Just needs to be rebuilt for Focal, and it's taking weeks & weeks (nothing new about that).

**Comment by 1beb** [comment]
@cement-head https://bugs.launchpad.net/ubuntu/+source/xorg-server/+bug/1931547 they built and provided links for hir/foc

**Comment by lspintzyk** [comment]
Please add:
deb http://ports.ubuntu.com/ubuntu-ports focal-proposed main
to your apt sources.list and to apt update & upgrade.

Or way couple days more until this will appear in main repo.

Issue is fixed.

**Comment by 1beb** [comment]
Can confirm it's working on Pop 21.04 on Wayland as well.

- Upgraded
- Uninstalled displaylink
- Reboot
- Installed displaylink (5.4) 

Poof! Monitors popped on! Yay!

**Comment by bolek2000** [comment]
On Pop 20.04:

- Removed  package holds for downgraded xserver and mesa
- Upgraded to newest versions
- Removed DL 5.3
- rebooted into kernel 5.11
- Installed downloaded xserver packages
`sudo dpkg -i xserver-common_1.20.11-1ubuntu1~20.04.2_all.deb xserver-xorg-core_1.20.11-1ubuntu1~20.04.2_amd64.deb xserver-xorg-legacy_1.20.11-1ubuntu1~20.04.2_amd64.deb`
- installed DL 5.4

For me the external display is now working again, but everything is responding very slowly when using the display link. So when I type, the letters are showing up a couple of seconds later on the screen. When I change windows by clicking it takes several seconds as well. If I remember right I had this behavior before long time ago, but I don't remember, if there was a solution...

**Comment by cement-head** [comment]
Fixed for me as well running Ubuntu 20.04 LTS on System76 Galp-3c using System76 PPA.

```
$ uname -a
Linux ######## 5.11.0-7614-generic #15~1622578982~20.04~383c0a9~dev-Ubuntu SMP Wed Jun 2 00:50:47 U x86_64 x86_64 x86_64 GNU/Linux

$ glxinfo | grep OpenGL
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) UHD Graphics 620 (WHL GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 21.0.0
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 4.6 (Compatibility Profile) Mesa 21.0.0
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 21.0.0
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
OpenGL ES profile extensions:
```
